code/parse/parselo.cpp

```diff
@@ -1665,30 +1666,94 @@ int maybe_convert_foreign_character(int ch)
 }
 
 // Goober5000
-void maybe_convert_foreign_characters(char *line)
+// Yarn - The capacity of out must be at least the value returned by
+// get_converted_string_length(in) (plus one if add_null is true).
+// Returns the number of characters written to out.
+size_t maybe_convert_foreign_characters(char *in, char *out, bool add_null)
```

Comment:
Just using SCP_strings would be better, I agree, but I don't think I have the experience necessary to determine the best way to convert the strings being passed to this function to SCP_string; those same strings get passed to or are generated from char*-only functions, so it looks like the conversion would involve more than just minor changes. If you or someone else can figure it out, then I will change my branch accordingly.